demos/data/scripts/generator/PersonGen.py

```diff
@@ -0,0 +1,438 @@
+            call_date = self.fake.date_time_between(start_date=start_date)
+            call_time = call_date + timedelta(hours=self.random.randint(0, 23), minutes=self.random.randint(0, 59), seconds=self.random.randint(0, 59))
+            duration = self.random.randint(1, 3600)  # Call duration in seconds, from 1 sec to 1 hour
+            
+            call_logs.append({
+                "caller": caller,
+                "callee": callee,
+                "call_date": call_date.strftime('%Y-%m-%d'),
+                "call_time": call_time.strftime('%H:%M:%S'),
+                "duration_sec": duration
+            })
+        
+        return pd.DataFrame(call_logs)
+
+    def generate_non_affiliated_call_logs(
+            self,
+            people_df: pd.DataFrame,
+            call_logs_df: pd.DataFrame,
+            num_calls: int = 500,
+            start_date: str = '-1y'
+            ) -> pd.DataFrame:
+        """
+        Generate call logs for non-affiliated individuals, simulating everyday calls.
+
+        :param people_df: DataFrame of people with affiliations.
+        :param call_logs_df: DataFrame of call logs to append to.
+        :param num_calls: Number of calls to generate among non-affiliated individuals.
+        :return: Updated DataFrame with non-affiliated call logs.
+        """
+        # Filter for non-affiliated individuals
+        non_affiliated_people = people_df[people_df['affiliation'] == 'None']
+        
+        # Generate call logs
+        for _ in range(num_calls):
+            if len(non_affiliated_people) > 1:
+                caller, callee = non_affiliated_people.sample(n=2, replace=False)['phone_number'].values
+                self.add_call_log(call_logs_df, caller, callee, start_date)
+
+        return call_logs_df
+
+    def generate_affiliated_call_logs(
+            self,
+            people_df: pd.DataFrame,
+            call_logs_df: pd.DataFrame,
+            num_affiliated_calls: int = 100,
+            leader_call_percentage: float = 0.05,
+            start_date: str = '-1y'
+            ) -> pd.DataFrame:
+        """
+        Generate call logs with a focus on gang affiliations, including both intra-gang and inter-gang communications.
+        
+        :param people_df: DataFrame of people with affiliations.
+        :param call_logs_df: Existing DataFrame of call logs to append to.
+        :param num_affiliated_calls: Number of additional affiliated calls to generate.
+        :param leader_call_percentage: Percentage of calls that should be between gang leaders (inter-gang calls).
+        :return: Updated DataFrame with affiliated call logs.
+        """
+        affiliated_people = people_df[people_df['affiliation'] != 'None']
+        affiliated_groups = affiliated_people['affiliation'].unique()
+        
+        leader_calls = int(num_affiliated_calls * leader_call_percentage)
+        gang_calls = num_affiliated_calls - leader_calls
+
+        # Generate intra-gang calls
```

Comment:
neato

i think this is similar to above... maybe for each gang:

- generate a general social network within the gang
- inject the leader or hierarchy or cell structure you want
- add some random overlap between the gang, other gangs, and unaffiliated

Reply:
(+ comment on burners)

Reply:
Yea was thinking about how to do this wrt the intra-gang call logs , for example only a leader would likely call to another gangs leaders